neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/internal/AllContentsIterator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2013-2018 Atlanmod, Inria, LS2N, and IMT Nantes.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v2.0 which accompanies
+ * this distribution, and is available at https://www.eclipse.org/legal/epl-2.0/
+ */
+
+package fr.inria.atlanmod.neoemf.core.internal;
+
+import fr.inria.atlanmod.neoemf.core.PersistentEObject;
+
+import org.eclipse.emf.common.util.AbstractTreeIterator;
+import org.eclipse.emf.ecore.EObject;
+
+import java.util.Iterator;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * An {@link Iterator} that iterates over all content of a {@link PersistentEObject}.
+ */
+@ParametersAreNonnullByDefault
+public class AllContentsIterator extends AbstractTreeIterator<EObject> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This class has 6 parents which is greater than 5 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)